packages/flutter_tools/lib/src/commands/create_base.dart

```diff
@@ -456,6 +459,27 @@ abstract class CreateBase extends FlutterCommand {
     }
 
     if (boolArg('pub')) {
+      final Environment environment = Environment(
+        artifacts: globals.artifacts,
+        logger: globals.logger,
+        cacheDir: globals.cache.getRoot(),
+        engineVersion: globals.flutterVersion.engineRevision,
+        fileSystem: globals.fs,
+        flutterRootDir: globals.fs.directory(Cache.flutterRoot),
+        outputDir: globals.fs.directory(getBuildDirectory()),
+        processManager: globals.processManager,
+        platform: globals.platform,
+        projectDir: project.directory,
+        generateDartPluginRegistry: true,
+      );
+
+      // Generate the l10n synthetic package that will be injected into the
+      // package_config in the call to pub.get() below.
+      await generateLocalizationsSyntheticPackage(
+        environment: environment,
+        buildSystem: globals.buildSystem,
+      );
```

Comment:
I don't know this system that well, but can't we just call the same code that `flutter packages get` calls, rather than having to more or less duplicate it here?

Reply:
I'm also not too familiar with it. That code is inside `PackagesGetCommand`. I don't know if it's weird to create/call another command from inside this command. If so, we could perhaps make a function that does both the generation+pub get and call it from both.

As far as I can see, there aren't many differences in the `pub get` call (at least that can't be handled with some arguments), although the one in the `PackagesGetCommand` does also record timings - I'm not sure whether that should/shouldn't be included when it's being called by `create`.

@christopherfujino are you more familiar with this - do you have an opinion on the best way to share this?

Reply:
No, not more familiar, I'm catching up with the source issue before I have an opinion...

Reply:
we can pull this out into a shared function, although, this is essentially a one line addition, of `await generateLocalizationsSyntheticPackage(...)`, it just looks longer because we're creating the `Environment` object on another line, which is forwarding a lot of globals. I think this code LGTM as is.